aptos-move/mvhashmap/src/versioned_data.rs

```diff
@@ -27,7 +28,7 @@ enum EntryCell<V> {
     /// has: 1) Incarnation number of the transaction that wrote the entry (note
     /// that TxnIndex is part of the key and not recorded here), 2) actual data
     /// stored in a shared pointer (to ensure ownership and avoid clones).
-    Write(Incarnation, Arc<V>),
+    Write(Incarnation, Arc<(V, Option<MoveTypeLayout>)>),
```

Comment:
So we do store in arc actually! I think the API should only allow to get a reference to this layout, what do you think? To avoid clones

Reply:
Are you suggesting to use `Arc<(V, Arc<Option<MoveTypeLayout>>)>`, or `(Arc<V>, Arc<Option<MoveTypeLayout>>)`? 
The second one looks better to me.

Reply:
Even with current signature, you can return reference to MoveTypeLayout?

Let's not have two arcs